doc/sphinx/language/core/basic.rst

```diff
@@ -0,0 +1,428 @@
+================================
+Basic vocabulary and conventions
+================================
+
+This section provides some basic vocabulary and conventions to read
+this manual.
+
+We start by explaining the syntax and lexical conventions used in this
+manual.
+
+Then, we explain the basic notions of the Coq language.  In
+particular, we provide a short glossary defining the most commonly
+used vocabulary.  Other glossary terms are defined throughout the
+manual and the reader may refer to the :ref:`glossary index
+<glossary_index>` for a more complete reference.
+
+Syntax and lexical conventions
+------------------------------
+
+Syntax conventions
+~~~~~~~~~~~~~~~~~~
+
+The grammar of Coq is an extensible, non-context-free grammar.  The
```

Comment:
I think it's a stretch to call it context-free at any given point.  A context-free presentation would have the grammar be modified at every single `Definition`, because you can bind a scope to a type so that scope is automatically opened for arguments of that to for any definition, and you can define a scope to automatically open a new custom entry with completely separate grammar rules.  So your CFG would need to have a separate rule for every definition in scope.

Reply:
Simple lookahead doesn't make the grammar non context free.

Reply:
No more reference to context-free grammars.